invisible_cities/cities/beersheba_test.py

```diff
@@ -0,0 +1,185 @@
+    beersheba(**conf)
+
+    tables = ( "MC/extents"  , "MC/hits"     , "MC/particles", "MC/generators",
+               "DECO/Events" ,
+               "Run/events"  , "Run/runInfo" )
+
+    with tb.open_file(true_out)  as true_output_file:
+        with tb.open_file(PATH_OUT) as      output_file:
+            for table in tables:
+                got      = getattr(     output_file.root, table)
+                expected = getattr(true_output_file.root, table)
+                assert_tables_equality(got, expected)
+
+
+def test_beersheba_exact_result3d(ICDIR, ICDATADIR, PSFDIR, config_tmpdir):
+    PATH_IN     = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC.h5")
+    true_out    = os.path.join(ICDATADIR, "exact_Kr_deconvolution3d_with_MC.h5")
+    PATH_OUT    = os.path.join(config_tmpdir, "beersheba_MC.h5")
+    config_path = os.path.join(ICDIR, "config/beersheba.conf")
+    conf        = configure(f'dummy {config_path}'.split())
+    nevt_req    = all_events
+    conf.update(dict(files_in                     = PATH_IN                   ,
+                     file_out                     = PATH_OUT                  ,
+                     event_range                  = nevt_req                  ,
+                     deconv_params     = dict(
+                                         q_cut           =              10,
+                                         drop_dist       =      [10., 10.],
+                                         psf_fname       =          PSFDIR,
+                                         e_cut           =            1e-3,
+                                         n_iterations    =              10,
+                                         iteration_tol   =            0.01,
+                                         sample_width    =  [10., 10., 2.],
+                                         bin_size        =  [ 1.,  1., 1.],
+                                         energy_type     =             'E',
+                                         diffusion       = (1.0, 1.0, 0.3),
+                                         deconv_mode     =         'joint',
+                                         n_dim           =               3,
+                                         cut_type        =           'rel',
+                                         inter_method    = 'linear')))
+    beersheba(**conf)
+
+    tables = ( "MC/extents"  , "MC/hits"     , "MC/particles", "MC/generators",
+               "DECO/Events" ,
+               "Run/events"  , "Run/runInfo" )
+
+    with tb.open_file(true_out)  as true_output_file:
+        with tb.open_file(PATH_OUT) as      output_file:
+            for table in tables:
+                got      = getattr(     output_file.root, table)
+                expected = getattr(true_output_file.root, table)
+                assert_tables_equality(got, expected)
+
+
+def test_beersheba_empty_input_file(ICDIR, ICDATADIR, PSFDIR, config_tmpdir):
```

Comment:
I think this was basically because in some cities (don't remember which), before, the city crashed when trying to run on an empty file and it caused the data processing daemon in Canfranc to stop working due to that.    So this basically checks that this does not happen when running the city.